### PR TITLE
[CI] Add CMake format check

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

1. Use cmake-format to check cmake files, including CMakeLists.txt & *.cmake.
2. Fix the format issues.
3. Refactor license check.

Example to use the format tool:
```
apt install python3-pip -y
pip3 install --user cmake-format
cmake-format --first-comment-is-literal True --in-place cpp/velox/CMakeLists.txt
```

### CODE REVIEW

**Comment by PHILO-HE** [comment]
I tried to put code format check for cpp/cpp-ch together and then will remove `ch_code_style.yml`. But it's strange that if we do that (i.e., by removing https://github.com/apache/incubator-gluten/pull/5941/commits/200e183a624f405e96709510c267aea91bc0b459), clang-format will report some error for some CH code. 
See https://github.com/apache/incubator-gluten/actions/runs/9346082625/job/25720197384.

And I tried to use clang-format-15 in local, same error is reported. @liuneng1994, could you take a look?

File: .github/workflows/code_style.yml [hunk 1]
```diff
@@ -14,43 +14,56 @@
...
+          pip3 install --user cmake-format
+      - uses: actions/checkout@v4
+      - name: Check CMake format
         run: |
-          git fetch --recurse-submodules=no origin main ${{github.event.pull_request.base.sha}}
-          pip install regex
-          dev/check.sh ${{github.event.pull_request.base.sha}}
+          git config --global --add safe.directory $GITHUB_WORKSPACE
+          cd $GITHUB_WORKSPACE/
+          fileList=$(find ./cpp ./cpp-ch -name CMakeLists.txt -o -name *.cmake)
```
Comment (PHILO-HE) [inline .github/workflows/code_style.yml:60]: @zzcclp, I'm adding to CI job to format CMake code (CMakeLists.txt & *.cmake), which also covers cpp-ch module. Please take a look to see if it's ok to you. Thanks!

**Comment by zzcclp** [comment]
@liuneng1994 @baibaichen please help to review and check, thanks.

**Review by ulysses-you — DISMISSED** [review]
lgtm

**Comment by PHILO-HE** [comment]
@liuneng1994, do you have any comment?

**Comment by liuneng1994** [comment]
> @liuneng1994, do you have any comment?

There is generally no problem, but subjectively, the effect of cmake-format seems to make some places more messy.

File: cpp-ch/CMakeLists.txt [hunk 2]
```diff
@@ -27,93 +29,93 @@ message("CH_COMMIT=${CH_COMMIT}")
...
-        -DENABLE_BCRYPT=OFF
-        -DENABLE_LDAP=OFF
-        -DENABLE_MSGPACK=OFF
-        -DUSE_REPLXX=OFF
-        -DENABLE_CLICKHOUSE_ALL=OFF
-        -DCOMPILER_FLAGS='-fvisibility=hidden -fvisibility-inlines-hidden'
-        -S ${CH_SOURCE_DIR} -G Ninja -B ${CH_BINARY_DIR} &&
-        cmake --build ${CH_BINARY_DIR} --target libch\"
-        OUTPUT _build_ch)
+  add_custom_command(
```
Comment (liuneng1994) [inline cpp-ch/CMakeLists.txt:100]: It seems to be getting messier here.

Comment (PHILO-HE) [reply cpp-ch/CMakeLists.txt:100]: @liuneng1994, I think for most code, the formatting makes code look better. Maybe, some configuration can be applied in the future to tune the format.

**Review by liuneng1994 — APPROVED** [review]

**Review by rui-mo — APPROVED** [review]

**Pull request merged**